components/locid/src/macros.rs

```diff
@@ -196,13 +196,39 @@ macro_rules! langid {
 /// limitations (see [`Heap Allocations in Constants`]):
 ///
 /// ```compile_fail
-/// icu::locid::locale!("en-US-u-ca-ja");
+/// icu::locid::locale!("sl-IT-rozaj-biske-1994")
 /// ```
 /// Use runtime parsing instead:
 /// ```
-/// "en-US-u-ca-ja".parse::<icu::locid::Locale>().unwrap();
+/// "sl-IT-rozaj-biske-1994".parse::<icu::locid::Locale>().unwrap();
 /// ```
 ///
+/// Locales with multiple keys are not supported
+/// ```compile_fail
+/// icu::locid::locale!("en-US-u-ca-buddhist-ca-japanese");
+/// ```
+/// Use runtime parsing instead:
+/// ```
+/// "en-US-u-ca-buddhist-ca-japanese".parse::<icu::locid::Locale>().unwrap();
+/// ```
+///
+/// Locales with attributes are not supported
+/// ```compile_fail
+/// icu::locid::locale!("en-US-u-foobar-ca-buddhist");
+/// ```
+/// Use runtime parsing instead:
+/// ```
+/// "en-US-u-foobar-ca-buddhist".parse::<icu::locid::Locale>().unwrap();
+/// ```
+///
+/// Locales with single key but multiple types are not supported
+/// ```compile_fail
+/// icu::locid::locale!("en-US-u-ca-buddhist-japanese");
```

Comment:
Suggestion (here and below): More realistic example

```suggestion
/// icu::locid::locale!("en-US-u-ca-islamic-umalqura");
```